app/components/access_panels/other_listings_component.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module AccessPanels
+  class OtherListingsComponent < AccessPanels::Base
+    def render?
+      earthworks_url.present? || purls.present?
+    end
+
+    def earthworks_url
+      @earthworks_url ||= construct_earthworks_url
+    end
+
+    # @return [Array<String>] URLs representing links to PURL, whether specified in MARC or MODS, and de-duped, e.g. ["https://purl.stanford.edu/hj948rn6493"]
+    def purls
+      @purls ||= (mods_other_locations + marc_other_locations).uniq.find_all { |str| str.start_with?(Settings.PURL_EMBED_RESOURCE) }
+    end
+
+    # @return [Array<String>] a tags representing PURL links
+    def purl_links
+      # text of the link is the actual URL if there is more than one unique URL, so there aren't multiple links with the same title
+      return purls.map { |purl| link_to(purl, purl) } if purls.size > 1
+
+      # returned as an array for type consistency
+      Array(link_to('Stanford Digital Repository', purls.first))
+    end
+
+    private
+
+    # @return [Array<String>] URLs representing links to other locations, extracted from MODS metadata, e.g. ["https://purl.stanford.edu/hj948rn6493"]
+    def mods_other_locations
+      #   latter of those is used in app/views/catalog/record/_mods_bibliographic.html.erb (but has some clutter unlike the mockup, for this purpose)
+      Array(document['url_fulltext'])
+    end
+
+    # @return [Array<String>] URLs representing links to other locations, extracted from MARC metadata, e.g. ["https://purl.stanford.edu/hj948rn6493"]
+    def marc_other_locations
+      # the fetch with the default empty hash prevents errors for data without MARC representation
+      Array(document.fetch('marc_links_struct', {}).pluck('href'))
```

Comment:
```suggestion
      document.marc_links.managed_purls.map(&:href)
```

probably gets you the right PURL links. There's probably records that link to supplemental resources or other things that you probably don't want to link here?

Reply:
ah yeah, i suspected the non-PURL links might be a problem, which is why i did the combining and filtering in the `#purls` method.  but i like this more direct approach, will try it.